addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -336,16 +334,31 @@ static func get_built_in_categories(_class_name: String) -> Array[BlockCategory]
 			props = ["modulate"]
 
 		"RigidBody2D":
-			# On body entered
+			var b = BLOCKS["entry_block"].instantiate()
+			b.block_format = "On body entered [body: NODE_PATH]"
+			# HACK: Blocks refer to nodes by path but the callback receives the node itself;
+			# convert to path
+			b.statement = "func _on_body_entered(_body: Node):\n\tvar body: NodePath = _body.get_path()"
+			b.signal_name = "body_entered"
```

Comment:
It would be nice to have an "On body [body] exited" block too by connecting to the body_exited signal.

Reply:
Pushed an amended patch that defines that.